python/lsst/sims/featureScheduler/basis_functions.py

```diff
@@ -216,6 +216,162 @@ def __call__(self, indx=None):
         return result
 
 
+class HADecAltAzPatchBasisFunction(Base_basis_function):
+    """Build an AltAz mask using patches defined as hour angle, declination, altitude and azimuth limits.
+
+    """
+    def __init__(self, nside=default_nside, condition_features=None, survey_features=None,
+                 patches=({'ha_min': 2., 'ha_max': 22.,
+                           'alt_max': 88., 'alt_min': 55.,
+                           'dec_min': -90., 'dec_max': 90,
+                           'az_min': 0., 'az_max': 360.},)):
+        """
+        Parameters
+        ----------
+        patches : list(dict())
+            A list of dictionaries containing the keywords (ha_min, ha_min), (alt_min, alt_max), (dec_min, dec_max),
+            (az_min and az_max) to construct an AltAz mask. One can skip a pair of min/max value if that is not
+            part of the patch definition. For instance patches = [{'alt_min': 30., 'alt_max': 86.5}] is a valid entry,
+            whereas patches = [{'alt_max': 86.5}] is not.
+        """
+        if nside is None:
+            nside = utils.set_default_nside()
+
+        self.nside = nside
+        if survey_features is None:
+            self.survey_features = {}
+        if condition_features is None:
+            self.condition_features = dict()
+            self.condition_features['altaz'] = features.AltAzFeature(nside=nside)
+            self.condition_features['lmst'] = features.Current_lmst()
+
+        self.patches = patches
+        ra, dec = utils.ra_dec_hp_map(self.nside)
+        self.ra_hours = ra*12./np.pi % 24.
+        self.dec_deg = dec*180./np.pi
+
+    def __call__(self, indx=None):
+        result = np.empty(hp.nside2npix(self.nside), dtype=float)
+        result.fill(hp.UNSEEN)
+
+        ha = (self.condition_features['lmst'].feature - self.ra_hours) % 24.
+
+        for patch in self.patches:
+            if 'ha_min' in patch and 'ha_max' in patch:
+                ha_mask = np.bitwise_or(ha <= patch['ha_min'],
+                                        ha >= patch['ha_max'])
+            else:
+                ha_mask = np.ones(hp.nside2npix(self.nside), dtype=bool)
+
+            if 'dec_min' in patch and 'dec_max' in patch:
```

Comment:
Since the dec map doesn't change, you could move the `dec_mask` calculation to the __init__ and do it just once. You could add an RA mask to the __init__ as well since it's the lone coordinate left out.